### PR TITLE
[Instrumentation.EntityFrameworkCore] Add Filter public API to enable filtering

### PR DESCRIPTION
**Pull request by akoken**
## Changes

This PR introduces the addition of a `Filter` public API, which enables filtering out activities based on the properties of the
`IDbCommand` object being executed. The `Filter` API provides users with the ability to apply custom filters to only collect specific activities, enhancing the flexibility and versatility of the system.

For significant contributions please make sure you have completed the following items:

* [X] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by Kielek** [comment]
Did you considered using`EnrichWithIDbCommand`? It should allow you to modify activity in context of IDbCommand.

**Comment by akoken** [comment]
Yes, we can modify activity with `EnrichWithIDbCommand`, but I don't think it's a right place to do that. `Encrich` and `Filter` should have their own purpose.

**Comment by Kielek** [comment]
@akoken, I have checked it internally with other maintainers.

Please consider following `Filter` option `Func<string, IDbCommand, bool>` instead of `Func<IDbCommand, bool>`.

The string parameter could be potentially providerName.

@utpilla, @alanwest FYI

**Comment by akoken** [comment]
@Kielek, I added `providerName` to the filter function and fixed the markdown issues.

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.EntityFrameworkCore/EntityFrameworkInstrumentationOptions.cs [hunk 1]
```diff
@@ -44,4 +44,25 @@ public class EntityFrameworkInstrumentationOptions
     /// <para><see cref="IDbCommand"/>: db command to allow access to command.</para>
     /// </remarks>
     public Action<Activity, IDbCommand> EnrichWithIDbCommand { get; set; }
+
+    /// <summary>
+    /// Gets or sets a filter function that determines whether or not to
+    /// collect telemetry about a command.
```
Comment (utpilla) [inline src/OpenTelemetry.Instrumentation.EntityFrameworkCore/EntityFrameworkInstrumentationOptions.cs:50]:
```suggestion
    /// collect telemetry about a command from a provider name.
```

**Review by utpilla — APPROVED** [review]

**Pull request merged**